KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutCoordinator.swift

```diff
@@ -121,6 +126,33 @@ extension KarhooNewCheckoutCoordinator: NewCheckoutRouter {
         viewController.modalPresentationStyle = .overFullScreen
         baseViewController.present(viewController, animated: true, completion: nil)
     }
+    
+    func routeToComment(title: String, comments: String) {
+        guard let presenter = presenter else {
+            assertionFailure("Presenter is missing")
+            return
+        }
+        let bottomSheetViewModel = KarhooBottomSheetViewModel(
+            title: title) { [weak self] in
+                self?.baseViewController.dismiss(animated: true, completion: nil)
+            }
+        let contentViewModel = KarhooBottomSheetCommentsViewModel(
+            initialValueForTextView: presenter.commentCellViewModel.getComment(),
+            viewSubtitle: UITexts.Booking.commentsSubtitle
+        ) { [weak self] comment in
+            self?.presenter?.commentCellViewModel.setComment(comment)
+            self?.baseViewController.dismiss(animated: true, completion: nil)
+        }
+        
+        let bottomSheet = KarhooBottomSheet(viewModel: bottomSheetViewModel) {
+            KarhooBottomSheetCommentsView(viewModel: contentViewModel)
+        }
+        
+        let viewController = UIHostingController(rootView: bottomSheet)
+        viewController.view.backgroundColor = UIColor.clear
+        viewController.modalPresentationStyle = .overFullScreen
+        baseViewController.present(viewController, animated: true, completion: nil)
```

Comment:
Could you please create some sort of builder, so we don't need to call these 20 lines of code every time a bottom sheet is needed? In this one file, we already have 3 invocations of this mechanism.

Reply:
I've already mentioned this and he tried, but for some reason they weren't being applied correctly. I agree that duplicating this logic every time is not sustainable. If you have time could you maybe give it a go? Doing it on the feature branch and then updating this one should be good enough so that we leave Bartek as the only contributor to this PR.